### PR TITLE
Improve custom panel support

### PR DESCRIPTION
**Pull request by balloob**
Add support for:
 - JS based custom panels
 - Embedding custom panels in iframes (optional) (allows React based panels)
 - Confirming with user if they want to load an external panel
 - Allow trusting individual external panels to be loaded without confirmation

Backend PR https://github.com/home-assistant/home-assistant/pull/14708
Starter kit example for React based panel: https://github.com/home-assistant/custom-panel-starter-kit-react

Config example:

```yaml
# Configuration.yaml example
panel_custom:
  - name: react-panel
    url_path: react-panel-prod
    sidebar_title: React Prod
    sidebar_icon: mdi:react
    js_url: https://s3.amazonaws.com/home-assistant-demos/panel-examples/react-demo.js
    embed_iframe: true
    config:
      name: World
```

image

### CODE REVIEW

File: src/panels/custom/ha-panel-custom.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    this._setProperties = null;
+    while (this.lastChild) {
+      this.remove(this.lastChild);
+    }
+
+    const tempA = document.createElement('a');
+    tempA.href = panel.config.html_url || panel.config.js_url;
+
+    if (!panel.config.trust_external && !['localhost', '127.0.0.1', location.hostname].includes(tempA.hostname)) {
+      if (!confirm(`Do you trust the external panel "${panel.config.name}" at "${tempA.href}"?
```
Comment (c727) [inline src/panels/custom/ha-panel-custom.js]: Afaik confirm does not work on mobile Safari or the iOS App

Comment (balloob) [reply src/panels/custom/ha-panel-custom.js]: https://stackoverflow.com/questions/18903525/alert-and-confirm-not-working-with-apple-mobile-web-app-capable

Fixed in iOS 7.0.3 according to this StackOverflow.

**Comment by c727** [comment]
`panel.config.config` is a bit odd, we will get a 2nd release with many forum posts about XXX is broken

**Comment by balloob** [comment]
Ok, fair point. I have an idea how to fix it. It means that we're going to keep panel config as `panel.config` and then have `panel.config._custom_panel` be our config.

**Pull request merged**